url6a9f553762_Bresiu_socket_io_tgz-pJ8-MainJ8/src/com/github/nkzawa/socketio/client/Url.java

```diff
@@ -0,0 +1,69 @@
+package com.github.nkzawa.socketio.client;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.regex.Pattern;
+
+public class Url {
+
+    private static Pattern PATTERN_HTTP = Pattern.compile("^http|ws$");
+    private static Pattern PATTERN_HTTPS = Pattern.compile("^(http|ws)s$");
+
+    private Url() {}
+
+    public static URL parse(String uri) throws URISyntaxException, MalformedURLException {
+        return parse(new URI(uri));
+    }
+
+    public static URL parse(URI uri) throws MalformedURLException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 36-49 into a separate method.